engine/runtime/src/main/java/org/enso/interpreter/runtime/scope/ModuleScope.java

```diff
@@ -262,9 +271,15 @@ public Optional<Type> getType(String name) {
     return Optional.ofNullable(types.get(name));
   }
 
-  /** @return the raw method map held by this module */
-  public Map<Type, Map<String, Function>> getMethods() {
-    return methods;
+  /** @return a map of methods for the given type */
+  public Map<String, Function> getMethodsForType(Type tpe) {
```

Comment:
What it would take to hide this (mutable) `Map<String, Function>`? It is the same encapsulation problem as with `getMethods()` - jut one level down?

Alternatively `Map<String,Function` could return an _immutable map_. However whatever operation is needed to perform a modification would still be encapsulated in `ModuleScope`. Such an encapsulation would make the code more understandable.

Reply:
It's possible, I think